### PR TITLE
add front matter and cleanup docs in V1 migration for docs site

### PR DESCRIPTION
**Pull request by piinthecloud**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR updates the following files by adding front matter, which is necessary for docs to display in developer.woocommerce.com/docs. 

The following files have had front matter added and the feedback sections removed:
`/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md`
`/third-party-developers/extensibility/checkout-block/available-filters.md`
`/third-party-developers/extensibility/checkout-block/available-filters/additional-cart-checkout-inner-block-types.md`
`/third-party-developers/extensibility/checkout-block/available-filters/cart-line-items.md`
`/third-party-developers/extensibility/checkout-block/available-filters/checkout-and-place-order-button.md`
`/third-party-developers/extensibility/checkout-block/available-filters/coupons.md`
`/third-party-developers/extensibility/checkout-block/available-filters/order-summary-items.md`
`/third-party-developers/extensibility/checkout-block/available-filters/totals-footer-item.md`
`/third-party-developers/extensibility/checkout-block/available-slot-fills.md`
`/third-party-developers/extensibility/checkout-block/dom-events.md`
`/third-party-developers/extensibility/checkout-block/how-checkout-processes-an-order.md`
`/third-party-developers/extensibility/checkout-block/integration-interface.md`
`/third-party-developers/extensibility/checkout-block/slot-fills.md`
`/third-party-developers/extensibility/checkout-payment-methods/filtering-payment-methods.md`
`/third-party-developers/extensibility/checkout-payment-methods/payment-method-integration.md`
`/third-party-developers/extensibility/hooks/migrated-hooks.md`

Additionally, `totals-footer-item.md`, `additional-checkout-fields.md`, `additional-cart-checkout-inner-block-types.md`, `cart-line-items.md`, `checkout-and-place-order-button.md`, `coupons.md`, `order-summary-items.md`, have been fully converted to markdown in order to allow for our ingestion plugin to work correctly. 

#### Files not included
This PR only includes files which will be added to the Developer Docs Sidebar Heading: **Cart and Checkout Blocks**

This PR does not include `plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/actions.md` and `plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/filters.md` as these are autogenerated by [this script ](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/bin/hook-docs/index.js) which is currently broken. 

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1.
2.
3.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by piinthecloud** [comment]
@opr how did you all manage to get around breaking rule MD024 in the past? For our ingestion plugin, those duplicate headers should't be a problem.

**Comment by opr** [comment]
> @opr how did you all manage to get around breaking rule MD024 in the past? For our ingestion plugin, those duplicate headers should't be a problem.

I actually didn't work around that rule, I updated the anchors. You can ignore like this though

`<!-- markdownlint-disable-line no-duplicate-heading -->`

**Comment by piinthecloud** [comment]
> I actually didn't work around that rule, I updated the anchors. You can ignore like this though

Hmm, something is off then. Because I did not change the anchors, and the rule is failing for multiple files. So I think maybe this was previously merged while ignoring that rule.

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ The following Coupon filters are available:
 -   [`showRemoveCouponNotice`](#showremovecouponnotice)
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md]: The docs header seems to be missing here:

```md
---
post_title: Cart and Checkout Filters - Coupons
menu_title: Coupons
tags: reference, checkout-available-filters
---
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ The following Coupon filters are available:
...
 
 ## `coupons`
-
+<!-- markdownlint-disable MD024 -->
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md]: I see that for other files, we're placing the lint rule below the docs header and above the docs title. For consistency-reasons, I suggest keeping the same format across all files.

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/totals-footer-item.md
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/totals-footer-item.md:1]: When applying `<!-- markdownlint-disable MD024 -->` to this file, we don't need to overwrite the duplicate titles and keep this doc consistent with the other docs.

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/how-checkout-processes-an-order.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+---
+post_title: How The Checkout Block Processes an Order
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/how-checkout-processes-an-order.md]: Based on https://github.com/woocommerce/woocommerce/blob/trunk/docs/quality-and-best-practices/grammar-punctuation-capitalization.md#capitalization, we should _"Use sentence case (not title case) for docs titles and subheadings."_ I suggest using sentence case for all docs titles to keep them consistent.

I also noticed that some docs titles are prefixed with `Cart and Checkout - ` while others are not. What are your thoughts on prefixing all or none, for consistency-reasons?

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @piinthecloud. I left a few questions and suggestions below. Please let me know in case something I mentioned should be unclear.

**Comment by nielslange** [comment]
> Hmm, something is off then. Because I did not change the anchors, and the rule is failing for multiple files. So I think maybe this was previously merged while ignoring that rule.

@piinthecloud, I wanted to clarify and add to our earlier discussion in p1719848980552729-slack-C8X6Q7XQU, especially to maintain transparency.

When we lint Markdown files in our CI, we use the config file located at `docs/.markdownlint.json`. However, for the `plugins/woocommerce-blocks` directory, the configuration was previously set to use `plugins/woocommerce-blocks/.markdownlint.json`. Additionally, there's a general config file in the root of the monorepo, `.markdownlint.json`, and all three files have slightly different rules, which could explain the discrepancies you're seeing.

@woocommerce/vortex is aware of these differences. As a temporary solution, similar to what you've done with the current lint errors, you can bypass specific lint rules in the docs by adding the following line under the docs header:

```md
<!-- markdownlint-disable MD024 -->
```

**Comment by piinthecloud** [comment]
Comments addressed @nielslange! Thanks for the keen eye

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/how-checkout-processes-an-order.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+---
+post_title: How the checkout block processes an order
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/how-checkout-processes-an-order.md]:
```suggestion
post_title: Cart and Checkout - How the Checkout block processes an order
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+---
+post_title: Additional Checkout Fields
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md]:
```suggestion
post_title: Cart and Checkout - Additional Checkout Fields
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/additional-cart-checkout-inner-block-types.md [hunk 6]
```diff
@@ -1,3 +1,10 @@
+---
+post_title: Cart and Checkout Filters - Inner Block Types
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/additional-cart-checkout-inner-block-types.md]:
```suggestion
post_title: Cart and Checkout - Inner Block Types
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/cart-line-items.md [hunk 7]
```diff
@@ -1,3 +1,11 @@
+---
+post_title: Cart and Checkout Filters - Cart Line Items
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/cart-line-items.md]:
```suggestion
post_title: Cart and Checkout - Cart Line Items
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/checkout-and-place-order-button.md [hunk 8]
```diff
@@ -1,3 +1,9 @@
+---
+post_title: Cart and Checkout Filters - Checkout and Place Order Button
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/checkout-and-place-order-button.md]:
```suggestion
post_title: Cart and Checkout - Checkout and Place Order Button
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md [hunk 9]
```diff
@@ -1,3 +1,11 @@
+---
+post_title: Cart and Checkout Filters - Coupons
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/coupons.md:2]:
```suggestion
post_title: Cart and Checkout - Coupons
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/order-summary-items.md [hunk 10]
```diff
@@ -1,3 +1,9 @@
+---
+post_title: Cart and Checkout Filters - Order Summary Items
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/order-summary-items.md:2]:
```suggestion
post_title: Cart and Checkout - Order Summary Items
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/totals-footer-item.md [hunk 11]
```diff
@@ -1,5 +1,10 @@
+---
+post_title: Cart and Checkout Filters - Totals Footer Item
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-filters/totals-footer-item.md]:
```suggestion
post_title: Cart and Checkout - Totals Footer Item
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-payment-methods/filtering-payment-methods.md [hunk 12]
```diff
@@ -1,3 +1,11 @@
+---
+post_title: Filtering Payment Methods in the Checkout block
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-payment-methods/filtering-payment-methods.md]:
```suggestion
post_title: Cart and Checkout - Filtering payment methods in the Checkout block
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-payment-methods/payment-method-integration.md [hunk 13]
```diff
@@ -1,3 +1,9 @@
+---
+post_title: Payment Method Integration for the Checkout Block
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-payment-methods/payment-method-integration.md]:
```suggestion
post_title: Cart and Checkout - Payment method integration for the Checkout block
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/migrated-hooks.md [hunk 14]
```diff
@@ -1,3 +1,9 @@
+---
+post_title: Cart and Checkout Filters - Legacy Hooks and Filters
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/migrated-hooks.md]:
```suggestion
post_title: Cart and Checkout - Legacy hooks
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/migrated-hooks.md [hunk 15]
```diff
@@ -1,3 +1,9 @@
...
+menu_title: Legacy Hooks and Filters
```
Comment (nielslange) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/migrated-hooks.md]:
```suggestion
menu_title: Legacy Hooks
```

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/migrated-hooks.md (hunk 14 above)
Comment (nielslange) [reply plugins/woocommerce-blocks/docs/third-party-developers/extensibility/hooks/migrated-hooks.md]: I'd either use `Legacy hooks` or `Legacy actions and filters`, as `actions` and `filters` are `hooks`.

**Review by nielslange — APPROVED** [review]
I left a few suggestions to ensure that the docs titles are consistent. Apart from that, the PR looks good. Thus, I'm pre-approving it. Thanks again for working on this, @piinthecloud.

**Pull request merged**